don't execute target/compile, target/linux/install & package/install if ecos is used
in addition, don't add libc as default dependency in ecos configs

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,7 @@ endif
 
 $(toolchain/stamp-install): $(tools/stamp-install)
 $(target/stamp-compile): $(toolchain/stamp-install) $(tools/stamp-install) $(BUILD_DIR)/.prepared
-$(package/stamp-compile): $(target/stamp-compile) $(package/stamp-cleanup)
+$(package/stamp-compile): $(target/stamp-compile) $(toolchain/stamp-install) $(package/stamp-cleanup)
 $(package/stamp-install): $(package/stamp-compile)
 $(target/stamp-install): $(package/stamp-compile) $(package/stamp-install)
 
diff --git a/include/package-defaults.mk b/include/package-defaults.mk
--- a/include/package-defaults.mk
+++ b/include/package-defaults.mk
@@ -5,7 +5,7 @@
 # See /LICENSE for more information.
 #
 
-PKG_DEFAULT_DEPENDS = +libc +SSP_SUPPORT:libssp +USE_GLIBC:librt +USE_GLIBC:libpthread
+PKG_DEFAULT_DEPENDS = +!PACKAGE_ecos:libc +SSP_SUPPORT:libssp +USE_GLIBC:librt +USE_GLIBC:libpthread
 
 ifneq ($(PKG_NAME),toolchain)
   PKG_FIXUP_DEPENDS = $(if $(filter kmod-%,$(1)),$(2),$(PKG_DEFAULT_DEPENDS) $(filter-out $(PKG_DEFAULT_DEPENDS),$(2)))
diff --git a/package/Makefile b/package/Makefile
--- a/package/Makefile
+++ b/package/Makefile
@@ -191,6 +191,8 @@ endif
 $(eval $(call stampfile,$(curdir),package,prereq,.config))
 $(eval $(call stampfile,$(curdir),package,cleanup,$(TMP_DIR)/.build))
 $(eval $(call stampfile,$(curdir),package,compile,$(TMP_DIR)/.build))
-$(eval $(call stampfile,$(curdir),package,install,$(TMP_DIR)/.build))
+ifeq ($(CONFIG_PACKAGE_ecos),)
+  $(eval $(call stampfile,$(curdir),package,install,$(TMP_DIR)/.build))
+endif
 
 $(eval $(call subdir,$(curdir)))
diff --git a/target/Makefile b/target/Makefile
--- a/target/Makefile
+++ b/target/Makefile
@@ -1,3 +1,4 @@
+
 # 
 # Copyright (C) 2007 OpenWrt.org
 #
@@ -8,12 +9,15 @@ curdir:=target
 
 $(curdir)/builddirs:=linux ltq-sdk sdk imagebuilder toolchain
 $(curdir)/builddirs-default:=linux
-$(curdir)/builddirs-install:=linux $(if $(CONFIG_SDK),sdk) $(if $(CONFIG_IB),imagebuilder) $(if $(CONFIG_MAKE_TOOLCHAIN),toolchain)
+$(curdir)/builddirs-install:=$(if $(CONFIG_PACKAGE_ecos),,linux) $(if $(CONFIG_SDK),sdk) $(if $(CONFIG_IB),imagebuilder) $(if $(CONFIG_MAKE_TOOLCHAIN),toolchain)
 
 $(curdir)/imagebuilder/install:=$(curdir)/linux/install
 
 $(eval $(call stampfile,$(curdir),target,prereq,.config))
-$(eval $(call stampfile,$(curdir),target,compile,$(TMP_DIR)/.build))
+ifeq ($(CONFIG_PACKAGE_ecos),)
+  $(eval $(call stampfile,$(curdir),target,compile,$(TMP_DIR)/.build))
+endif
+
 $(eval $(call stampfile,$(curdir),target,install,$(TMP_DIR)/.build))
 
 $($(curdir)/stamp-install): $($(curdir)/stamp-compile) 
